source/common/ratelimit/ratelimit_impl.h

```diff
@@ -16,9 +16,16 @@
 namespace Envoy {
 namespace RateLimit {
 
-class GrpcClientImpl : public Client, public Grpc::RpcChannelCallbacks {
+typedef Grpc::AsyncClient<pb::lyft::ratelimit::RateLimitRequest,
+                          pb::lyft::ratelimit::RateLimitResponse>
+    RateLimitAsyncClient;
+typedef std::unique_ptr<RateLimitAsyncClient> RateLimitAsyncClientPtr;
+
+typedef Grpc::AsyncRequestCallbacks<pb::lyft::ratelimit::RateLimitResponse> RateLimitAsyncCallbacks;
+
+class GrpcClientImpl : public Client, public RateLimitAsyncCallbacks {
```

Comment:
In looking at this code again, I realize that it's kind of dumb that we make a new "client" for every filter/request. Can you just drop a TODO in here while you are in here to optimize this at some point? (We should have thread local client per filter, and track outstanding requests).